### PR TITLE
Revise `ui_yeah` / `ui_nope`

### LINKED ISSUE #796: `ui_yeah` and `ui_nope` option choices and ordering
I like both of these functions and would like to use them in `ghclass` when users request potentially destructive actions. However, I'd like the option to control what choices are presented to the user and whether or not they are randomized.

Both of these behaviors were also unclear to me from the docs. 

Happy to propose a modified version of both if that would be helpful.

### PR DESCRIPTION
**Pull request by rundel**
Closes #796

Here is one possible starting point for enabling more control over `ui_yeah` and `ui_nope`. The goal here is to let the package author specific the exact options that will be shown for both the yes and no options and control whether they are randomly shuffled.

### CODE REVIEW

File: R/ui.R [hunk 1]
```diff
@@ -124,9 +124,43 @@ ui_warn <- function(x, .envir = parent.frame()) {
...
+#' ui_yeah("Do you like R?")
+#' ui_nope("Have you tried turning it off and on again?", n_yes = 1, n_no = 1)
+#' ui_yeah("Are you sure its plugged in?", yes = "Yes", no = "No", shuffle = FALSE)
+#' }
+NULL
+
+#' @rdname ui_question
 #' @export
-ui_yeah <- function(x, .envir = parent.frame()) {
+ui_yeah <- function(
```
Comment (jennybc) [inline R/ui.R]: Before this is done, you should run this through styler, because we format this differently.

File: R/ui.R [hunk 2]
```diff
@@ -137,21 +171,32 @@ ui_yeah <- function(x, .envir = parent.frame()) {
...
-  nays <- c("No way", "Not now", "Negative", "No", "Nope", "Absolutely not")
+  n_yes <- min(n_yes, length(yes))
+  n_no <- min(n_no, length(no))
 
-  qs <- c(sample(ayes, 1), sample(nays, 2))
-  ord <- sample(length(qs))
+  qs <- c(sample(yes, n_yes), sample(no, n_no))
+
+  if (shuffle)
+    qs <- sample(qs)
```
Comment (jennybc) [inline R/ui.R]: styler will put brackets around this.

File: R/ui.R [hunk 3]
```diff
@@ -137,21 +171,32 @@ ui_yeah <- function(x, .envir = parent.frame()) {
...
 #' @export
-ui_nope <- function(x, .envir = parent.frame()) {
-  !ui_yeah(x, .envir = .envir)
+ui_nope <- function(
+  x,
+  yes =  c("Yes", "Definitely", "For sure", "Yup", "Yeah", "I agree", "Absolutely"),
+  no  =  c("No way", "Not now", "Negative", "No", "Nope", "Absolutely not"),
+  n_yes = 1, n_no = 2, shuffle = TRUE,
+  .envir = parent.frame()
+) {
```
Comment (jennybc) [inline R/ui.R]: See styling comment from above.

File: R/ui.R [hunk 4]
```diff
@@ -124,9 +124,43 @@ ui_warn <- function(x, .envir = parent.frame()) {
...
+#'   when either a no option or no selection is made (canceled). `ui_nope` is the logical
+#'   opposite of `ui_yeah`.
+#' @name ui_question
+#' @examples
+#' \dontrun{
+#' ui_yeah("Do you like R?")
+#' ui_nope("Have you tried turning it off and on again?", n_yes = 1, n_no = 1)
+#' ui_yeah("Are you sure its plugged in?", yes = "Yes", no = "No", shuffle = FALSE)
+#' }
+NULL
```
Comment (jennybc) [inline R/ui.R]: I think this and the help from the rest of the `ui_*()` functions should link to each other, if we split them up.

File: R/ui.R (hunk 1 above)
Comment (rundel) [reply R/ui.R]: Sounds good, just ran things through and updated

File: R/ui.R (hunk 4 above)
Comment (rundel) [reply R/ui.R]: Makes sense, revisions include explicit links in the text as well as the inclusion of a user interface family

**Comment by rundel** [comment]
I'm happy to contribute tests, but I don't have any experience in working around the interactive issue. Any suggestions on spoofing `interactive()` and `menu()`?

**Comment by jennybc** [comment]
Thanks! Don't worry about the tests. I can pursue that myself.

**Pull request merged**